Loop/Managers/Alerts/AlertStore.swift

```diff
@@ -343,46 +350,59 @@ extension AlertStore {
         }
     }
 
-    func executeQuery(since date: Date, excludingFutureAlerts: Bool = true, now: Date = Date(), limit: Int, completion: @escaping (QueryResult<SinceDateFilter>) -> Void) {
+    public enum AlertQueryResult {
+        case success(QueryAnchor, [SyncAlertObject])
+        case failure(Error)
+    }
+
+    func executeQuery(fromQueryAnchor queryAnchor: QueryAnchor? = nil, since date: Date, excludingFutureAlerts: Bool = true, now: Date = Date(), limit: Int, completion: @escaping (AlertQueryResult) -> Void) {
         let sinceDateFilter = SinceDateFilter(predicateExpressionNotYetExpired: predicateExpressionNotYetExpired,
                                               date: date,
                                               excludingFutureAlerts: excludingFutureAlerts,
                                               now: now)
-        executeAlertQuery(from: QueryAnchor(filter: sinceDateFilter), limit: limit, completion: completion)
+        executeAlertQuery(fromQueryAnchor: queryAnchor, queryFilter: sinceDateFilter, limit: limit, completion: completion)
     }
 
-    func continueQuery<Filter: QueryFilter>(from anchor: QueryAnchor<Filter>, limit: Int, completion: @escaping (QueryResult<Filter>) -> Void) {
-        executeAlertQuery(from: anchor, limit: limit, completion: completion)
-    }
+    func executeAlertQuery(fromQueryAnchor queryAnchor: QueryAnchor?, queryFilter: QueryFilter? = nil, limit: Int, completion: @escaping (AlertQueryResult) -> Void) {
+        var queryAnchor = queryAnchor ?? QueryAnchor()
+        var queryResult = [SyncAlertObject]()
+        var queryError: Error?
 
-    private func executeAlertQuery<Filter: QueryFilter>(from anchor: QueryAnchor<Filter>, limit: Int, completion: @escaping (QueryResult<Filter>) -> Void) {
-        self.managedObjectContext.perform {
-            guard limit > 0 else {
-                completion(.success((anchor, [])))
-                return
-            }
+        guard limit > 0 else {
+            completion(.success(queryAnchor, []))
+            return
+        }
+
+        self.managedObjectContext.performAndWait {
```

Comment:
Just wondering, why did you switch to using `perform` instead of `performAndWait` here?  Isn't it better to not block the calling thread?

Reply:
The `RemoteDataServicesManager` implementation requires it (as it is currently designed). It could be changed by adding a DispatchGroup to the `RemoteDataServicesManager` function, but to be consistent it would require changing all of the stores and all of the code in `RemoteDataServicesManager`.